### PR TITLE
fix: CI failed

### PR DESCRIPTION
**Pull request by nic-chen**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues

___
### Bugfix
- Description
`Vars` in  struct `Route` uses interface{} as the type, which makes it impossible to omit an empty array.

As a result, after APISIX uses the new version of `lua-resty-radixtree`, The routes with empty array `vars` cannot be matched

- How to fix?

uses []interface{} instead. and also fix some small potential unstable issues

### CODE REVIEW

File: api/internal/core/entity/entity.go [hunk 1]
```diff
@@ -81,7 +81,7 @@ type Route struct {
 	Hosts           []string               `json:"hosts,omitempty"`
 	RemoteAddr      string                 `json:"remote_addr,omitempty"`
 	RemoteAddrs     []string               `json:"remote_addrs,omitempty"`
-	Vars            interface{}            `json:"vars,omitempty"`
+	Vars            []interface{}          `json:"vars,omitempty"`
```
Comment (Jaycean) [inline api/internal/core/entity/entity.go:84]: Why should the data format be modified here? Whether it has influence on the existing FE and BE logic.

Comment (starsz) [reply api/internal/core/entity/entity.go:84]: The same question as @Jaycean.

Comment (nic-chen) [reply api/internal/core/entity/entity.go:84]: @Jaycean  please take a look at `Bugfix Description`, I wrote the reason there.

Comment (Jaycean) [reply api/internal/core/entity/entity.go:84]: Sorry, my question. I didn't read the description carefully.

Comment (Jaycean) [reply api/internal/core/entity/entity.go:84]: cc @nic-chen 
FE CI failed:https://github.com/apache/apisix-dashboard/pull/1594/checks?check_run_id=2110905208
Remove the corresponding data: `x-apimix-vars ": []`

**Review by Jaycean — COMMENTED** [review]
This line of code also needs to be removed:

https://github.com/apache/apisix-dashboard/blob/21bff74080ab9c8dcb4de59db48d440c05bd6388/web/cypress/fixtures/export-route-dataset.json#L153

**Review by Jaycean — APPROVED** [review]
LGTM

**Review by liuxiran — APPROVED** [review]
LGTM for fe e2e changs

**Pull request merged**